CRM/Admin/Form/Preferences/Contribute.php

```diff
@@ -46,112 +46,127 @@ class CRM_Admin_Form_Preferences_Contribute extends CRM_Admin_Form_Preferences {
   );
 
   /**
-   * Process the form submission.
+   * Our standards for settings are to have a setting per value with defined metadata.
+   *
+   * Unfortunately the 'contribution_invoice_settings' has been added in non-compliance.
```

Comment:
Agreed, 'contribution_invoice_settings' is being stored in unusual way. Changing this now might cause problems for Sales tax or Deferred revenue since I believe they all rely on this setting.

Reply:
was  I too grumpy in my code comments :-)